opm/grid/cpgrid/CpGrid.cpp

```diff
@@ -75,81 +140,112 @@ CpGrid::scatterGrid(EdgeWeightMethod method, const std::vector<cpgrid::OpmWellTy
     static_cast<void>(transmissibilities);
     static_cast<void>(overlapLayers);
     static_cast<void>(method);
-#if HAVE_MPI
     if(distributed_data_)
     {
         std::cerr<<"There is already a distributed version of the grid."
                  << " Maybe scatterGrid was called before?"<<std::endl;
         return std::make_pair(false, std::unordered_set<std::string>());
     }
 
-    CollectiveCommunication cc(MPI_COMM_WORLD);
+#if HAVE_MPI
+    auto& cc = data_->ccobj_;
 
-    int my_num=cc.rank();
+    if (cc.size() > 1)
+    {
+        int my_num=cc.rank();
 #ifdef HAVE_ZOLTAN
-    auto part_and_wells =
-        cpgrid::zoltanGraphPartitionGridOnRoot(*this, wells, transmissibilities, cc, method, 0);
-    int num_parts = cc.size();
-    using std::get;
-    auto cell_part = std::get<0>(part_and_wells);
-    auto defunct_wells = std::get<1>(part_and_wells);
+        auto part_and_wells =
+            cpgrid::zoltanGraphPartitionGridOnRoot(*this, wells, transmissibilities, cc, method, 0);
+        using std::get;
+        auto cell_part = std::get<0>(part_and_wells);
+        auto defunct_wells = std::get<1>(part_and_wells);
+        auto exportList = std::get<2>(part_and_wells);
+        auto importList = std::get<3>(part_and_wells);
 #else
-    std::vector<int> cell_part(current_view_data_->global_cell_.size());
-    int  num_parts=-1;
-    std::array<int, 3> initial_split;
-    initial_split[1]=initial_split[2]=std::pow(cc.size(), 1.0/3.0);
-    initial_split[0]=cc.size()/(initial_split[1]*initial_split[2]);
-    partition(*this, initial_split, num_parts, cell_part, false, false);
-    const auto& cpgdim =  logicalCartesianSize();
-    std::vector<int> cartesian_to_compressed(cpgdim[0]*cpgdim[1]*cpgdim[2], -1);
-    for( int i=0; i < numCells(); ++i )
-    {
-        cartesian_to_compressed[globalCell()[i]] = i;
-    }
+        OPM_THROW(std::runtime_error, "Parallel runs depend on ZOLTAN. Please install!");
```

Comment:
if this is now a hard req, that is the job of the build system to enforce.

Reply:
The build system can only check whether Zoltan is there and MPI activated. How exactly should it check that we are running on more than one process?

Reply:
One could imaging a parallel program in a downstream module that runs a serial simulation on each MPI rank.

Reply:
yeah, sorry. i missed a bit of the context in the diff, this must indeed be a runtime check.